File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/beecrowd/1257/Array_Hash.c
#include <stdio.h>
#include <string.h>

void contar(int N){
    while(N > 0){
        int L = 0;
        scanf("%d", &L);
        getchar();

        int result = 0;
        for(int i = 0; i < L; i++){
            char line[51];
            fgets(line, sizeof(line), stdin);
            int len = strlen(line) - 1;

            for(int j = 0; j < len; j++){
                int valor = (line[j] - 'A') + i + j;
                result = result + valor; 
            }
        }

        printf("%d\n", result);
        N--;
    }
}

int main(void){
    int N = 0;
    int result = 0;

    scanf("%d", &N);
    contar(N);

   return 0;
}

File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/beecrowd/1257/Array_Hash.java
import java.util.*;

public class Array_Hash{
    public static void contar(int N, Scanner sc){
        while(N > 0){
            int L = sc.nextInt();
            sc.nextLine();

            int result = 0;
            for(int i = 0; i < L; i++){
                String line = sc.nextLine();

                for(int j = 0; j < line.length(); j++){
                    int valor = (line.charAt(j) - 'A') + i + j;
                    result = result + valor;
                }
            }

            System.out.println(result);
            N--;
        }
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();

        contar(N, sc);
        sc.close();
    }
}
